src/convertToObject.js

```diff
@@ -7,6 +7,29 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  const result = {};
+  const regex = /\n+|\t+/g;
+
+  sourceString
+    .replaceAll(regex, ' ')
+    .split(';')
+    .map((item) => item.trim())
+    .filter((item) => item !== '')
+    .map((item) => {
+      return item.split(':').map((x) => {
```

Comment:
suggest to reduce array iterations count.
after split(';') it can be just one iteration (use reduce method or if it is too hard - forEach will be enough).